### ISSUE TITLE
Request for Package: 'jdownloader'

### ISSUE THREAD

**Issue by colemickens** [NixOS/nixpkgs#41321]
Package this for NixOS: http://jdownloader.org

---

EDIT by @cole-h (copied from dupe issue https://github.com/NixOS/nixpkgs/issues/76568 because it may help with packaging):

**Metadata**

* homepage URL: http://www.jdownloader.org/ & http://beta.jdownloader.org/
* source URL: svn://svn.jdownloader.org/jdownloader & https://svn.jdownloader.org/projects
* license: **UNKNOWN, please see discussion below**
* platforms: unix, linux, darwin, win32

**Comment by PanAeon** [comment]
As a workaround, it provides a single jar file, command `java -jar JDownloader.jar`  successfully launches it on nixos.

**Comment by bosomt** [comment]
> As a workaround, it provides a single jar file, command `java -jar JDownloader.jar` successfully launches it on nixos.

works great ! thanks !

**Comment by bosomt** [comment]
it's not working after latest update ;(

       MyJDownloader Setup
|       Your 'My JDownloader' logins are not correct. 
|       Please check username/email and password!
|       Enter y -> Enter Logins
|       Enter n -> Exit JDownloader

**Comment by MicrosoftTakeover** [comment]
You can try JDownloader with Flatpak https://flathub.org/apps/details/org.jdownloader.JDownloader

**Comment by bosomt** [comment]
@MicrosoftTakeover thank but i will stay with remote web management, that tool is asking for:
718,2 MB will be downloaded in total.
2,3 GB more disk space will be used.
that's little bit too much ;)

**Comment by mkg20001** [comment]
Still important to me

Am Mo, 26. Jul 2021 um 07:25:06 -0700 schrieb stale[bot] 
***@***.***>:

**Comment by milahu** [comment]
> command `java -jar JDownloader.jar` successfully launches it on nixos.

now jdownloader has access to your clipboard and can make screenshots
it's a security risk, as jdownloader is closed source, while being advertised as "open source"

better security:
[docker](https://nixos.wiki/wiki/Docker) + jdownloader + vnc webinterface = https://hub.docker.com/r/jlesage/jdownloader-2/

> ```docker pull jlesage/jdownloader-2```
> 
> ```docker run -d --name=jdownloader-2 -p 5800:5800 -v /docker/appdata/jdownloader-2:/config:rw -v $HOME/Downloads/jd:/output:rw jlesage/jdownloader-2```
> 
> ```xdg-open http://localhost:5800/```

shell: ```docker exec -it jdownloader-2 sh```

i use `$HOME/Downloads/jd` so that jdownloader cannot see files in my `$HOME/Downloads`

> vnc webinterface

or use the official webinterface https://my.jdownloader.org/

**Comment by Krutonium** [comment]
> > command `java -jar JDownloader.jar` successfully launches it on nixos.
> 
> now jdownloader has access to your clipboard, screen, files, ... it's a security risk, as jdownloader is closed source, while being advertised as "open source"
> 
> better security: [docker](https://nixos.wiki/wiki/Docker) + jdownloader + vnc webinterface = https://hub.docker.com/r/jlesage/jdownloader-2/
> 
> > `docker pull jlesage/jdownloader-2`
> > `docker run -d --name=jdownloader-2 -p 5800:5800 -v /docker/appdata/jdownloader-2:/config:rw -v $HOME/Downloads/jd:/output:rw jlesage/jdownloader-2`
> > `xdg-open http://localhost:5800/`
> 
> shell: `docker exec -it jdownloader-2 sh`
> 
> i use `$HOME/Downloads/jd` so that jdownloader cannot see files in my `$HOME/Downloads`
> 
> > vnc webinterface
> 
> or use the official webinterface at [my.jdownloader.org](https://my.jdownloader.org/) you will need the [my jdownloader browser extension](https://my.jdownloader.org/apps/) to solve captchas

But it is open source? The source is on SVN and it's licensed as GPL. https://beta.jdownloader.org/developmentquicktutorial

**Comment by milahu** [comment]
> But it is open source?

no. the core is closed source
to build from source, you need JDClosed.jar (or similar)

**Comment by eribertto** [comment]
> > command `java -jar JDownloader.jar` successfully launches it on nixos.
> 
> now jdownloader has access to your clipboard, screen, files, ... it's a security risk, as jdownloader is closed source, while being advertised as "open source"
> 
> better security: [docker](https://nixos.wiki/wiki/Docker) + jdownloader + vnc webinterface = https://hub.docker.com/r/jlesage/jdownloader-2/
> 
> > `docker pull jlesage/jdownloader-2`
> > `docker run -d --name=jdownloader-2 -p 5800:5800 -v /docker/appdata/jdownloader-2:/config:rw -v $HOME/Downloads/jd:/output:rw jlesage/jdownloader-2`
> > `xdg-open http://localhost:5800/`
> 
> shell: `docker exec -it jdownloader-2 sh`
> 
> i use `$HOME/Downloads/jd` so that jdownloader cannot see files in my `$HOME/Downloads`
> 
> > vnc webinterface
> 
> or use the official webinterface at [my.jdownloader.org](https://my.jdownloader.org/) you will need the [my jdownloader browser extension](https://my.jdownloader.org/apps/) to solve captchas

Hi there, is this setup working in the latest NixOS release? I want to use jdownloader too but I'm new to NixOS coming from Arch/Fedora distros.

**Comment by milahu** [comment]
yeah, why not. you just need to enable [docker](https://nixos.wiki/wiki/Docker) and `nixos-rebuild switch`

**Comment by LDprg** [comment]
Porting [the arch pkgbuild](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=jdownloader2) to a nix shouldn't be so complicated. So maybe someone would like to do this and make a PR.

**Comment by PanAeon** [comment]
can we just close this issue if the source code is not available, and workaround exists? Otherwise, please, update the license as closed-source!

**Comment by colemickens** [comment]
I went ahead and updated the license so it doesn't indicate gpl3. To be honest, I have no idea what it's license is, I can't find source for it (I didn't actually try an SVN client, but the "updated" SVN URL seems to redirect->404 as well) and at this point the *ick*-factor is way too high for me to use/consider/touch jdownloader.

I will leave this open as I suspect some folks still want this, but I won't be addressing it, and I'd be fine if someone else took the liberty to close it.
